x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorPrivileges.java

```diff
@@ -114,14 +142,60 @@ public ElasticsearchSecurityException check(
             )) {
                 // Only check whether request is operator-only when user is NOT an operator
                 logger.trace("Checking operator-only violation for user [{}] and action [{}]", user, action);
-                final OperatorOnlyRegistry.OperatorPrivilegesViolation violation = operatorOnlyRegistry.check(action, request);
+                final OperatorPrivilegesViolation violation = operatorOnlyRegistry.check(action, request);
                 if (violation != null) {
                     return new ElasticsearchSecurityException("Operator privileges are required for " + violation.message());
                 }
             }
             return null;
         }
 
+        @Override
+        public RestResponse checkRestFull(RestHandler restHandler, ThreadContext threadContext) {
+            if (false == shouldProcess()) {
+                return null;
+            }
+            if (false == AuthenticationField.PRIVILEGE_CATEGORY_VALUE_OPERATOR.equals(
+                threadContext.getHeader(AuthenticationField.PRIVILEGE_CATEGORY_KEY)
+            )) {
+                // Only check whether request is operator-only when user is NOT an operator
+                if (logger.isTraceEnabled()) {
+                    Authentication authentication = threadContext.getTransient(AuthenticationField.AUTHENTICATION_KEY);
+                    final User user = authentication.getEffectiveSubject().getUser();
+                    logger.trace(
+                        "Checking for full REST restriction for user [{}] and routes [{}]",
+                        user,
+                        restHandler.routes().stream().map(RestHandler.Route::getPath).collect(Collectors.joining(","))
+                    );
```

Comment:
hmm... this is only a trace level message and neither RestRequest nor RestHandler implement toString() so they can't really be used for logging.  If this is trace message as-is (or minor change) isn't sufficient or misleading then maybe I should just remove it entirely. It probably isn't worth introducing addtional code and is only here since I found it useful during development. I think it is OK as-is since the enforcement will also be (string) path based which doesn't care about the HTTP verb and if any paths from this output matches something defined in the implementation to enforce then the restriction will be enforced.

Reply:
...actually while working on the followup PR I found another reason (better error response code) to pass in the RestRequest via the API. I will adjust this PR to pass that and use the restRequest.path() in the trace logger.